website/docusaurus.config.js

```diff
@@ -0,0 +1,141 @@
+const darkCodeTheme = require('prism-react-renderer/themes/dracula');
+const path = require('path');
+
+/**
+ * Markdown documentation in docs/ references repository directories
+ * and files. Relative paths are used so manually written documentation 
+ * can be read on GitHub entirely. 
+ * For our own documentation website we want to transform these
+ * relative links to matching GitHub link. 
+ * This will ensure consistent experience no matter where you
+ * start browsing the documentation.
+ * 
+ * Keep in mind docs/ is copied to website/docs/.
+ * Needed to work around an issue with TypeDoc x Docusaurus plugin.
+ * 
+ * The algorithm roughly is:
+ * - Visit all relative links within markdown documents.
+ * - Try resolving it relative to repo root. Some magic involved here. 
+ * - If it matches one of the hardcoded prefixes make it a GitHub link.
+ */
+function remarkPluginFixLinksToRepositoryArtifacts() {
+  /** @type {import('unified').Transformer} */
+  const transformer = async (ast, file) => {
+    // Package does not support require es modules.
+    // Easiest workaround I found.
+    let { visit } = await import('unist-util-visit');
+    visit(ast, 'link', (node) => {
+      /** @type string */
+      let link = node.url;
+      if (link.startsWith('http://') || link.startsWith('https://')) {
+        return;
+      }
+      let repoRoot = path.resolve(__dirname, '..');
+      let absolute = path.resolve(repoRoot, file.dirname, link);
+      let relative = path.relative(__dirname, absolute);
+      if (relative.startsWith('src')
+        || relative.startsWith('cursorless-snippets')
+        || relative.startsWith('.github')) {
+        const repoLink = 'https://github.com/cursorless-dev/cursorless-vscode/tree/main/';
+        const linkToRepositoryArtifact = repoLink.concat(relative);
+        node.url = linkToRepositoryArtifact;
+      }
+    });
+  };
+  return transformer;
+};
+
+/** @type {import('@docusaurus/types').Config} */
+const config = {
+  title: 'Cursorless',
+  tagline: 'Never use the cursor again',
+  // GitHub pages url
+  url: 'https://cursorless-dev.github.io/',
+  baseUrl: '/cursorless-vscode/',
+  onBrokenLinks: 'throw',
+  onBrokenMarkdownLinks: 'warn',
+  favicon: 'img/favicon.ico',
+  organizationName: 'pokey',
+  projectName: 'cursorless',
+
+  staticDirectories: [
+    'static',
+  ],
+
+  plugins: [
+    [
+      'docusaurus-plugin-typedoc',
+
+      // Plugin / TypeDoc options
+      {
+        entryPoints: ['../src/'],
+        entryPointStrategy: 'expand',
+        readme: 'none',
+        tsconfig: '../tsconfig.json',
+        out: 'contributing/api',
+        plugin: [
+          'typedoc-plugin-rename-defaults', 
+          'typedoc-plugin-mdn-links',
+          // typedoc generates using <internal> tag that is not closed. 
+          // MDX in docusaurus does not like that and reports unclosed jsx tag.
+          // 'typedoc-plugin-missing-exports'
```

Comment:
how come this one is commented out?

Reply:
I tried to explain it in the comment. Basically when using this plugin `<internal>` text in markdown is emitted. Docusaurus relies on MDX so it treats that string as a unclosed tag. I wasn't able to come up with an easy fix so just kept this as a comment. I'm actually not sure how much need we have for this plugin

Reply:
Ok, maybe let's remove this block but just file an issue to capture this one, which we'll probably close soon because as you say, it's not much of a problem.

Reply:
This actually got resolved. That plugin takes an option that can customize internal namespace name